File: projects/native-apps/ios/App/ShareArtifacts.swift
import Foundation

enum ShareArtifacts {
	static func purge(appGroupId: String) {
		guard let container = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: appGroupId) else { return }
		UploadJobStore(containerURL: container).purgeAll()
		UnseenSave(containerURL: container).clear()
		DiscoveryHTTPCache.purge(containerURL: container)
	}

	static func forgetReaderChoices(appGroupId: String) {
		AppGroupContainer.entitled(appGroupId: appGroupId)
			.map(ShareTarget.init(container:))?
			.forget()
		guard let defaults = UserDefaults(suiteName: appGroupId) else { return }
		LastViewedReadlist(defaults: defaults).forget()
	}
}

File: projects/native-apps/android/app/src/main/kotlin/com/readplace/android/core/ShareArtifacts.kt
package com.readplace.android.core

interface PurgeableUploadQueue {
	fun purgeAll()
}

class ShareArtifacts(
	private val jobs: PurgeableUploadQueue,
	private val unseenSave: UnseenSave,
	private val discoveryCache: DiscoveryHttpCache,
) {
	fun purge() {
		jobs.purgeAll()
		unseenSave.clear()
		discoveryCache.purge()
	}
}
